### ISSUE TITLE
Replace the proprietary Google location lib with a FOSS microG implementation

### ISSUE THREAD

**Issue by pastk** [organicmaps/organicmaps#9916]
- related to #9909

https://github.com/organicmaps/organicmaps/pull/9575 has added the microG implementation into the F-Droid builds already.
If it works well and there are no issues then we can gradually replace it in Web and Google flavors too.
And eventually get rid of the only proprietary bit left in those builds.

**Comment by pastk** [comment]
At the moment the microG implementation can't properly detect presence of Google Service on a device (see https://github.com/microg/GmsCore/issues/2309).

This means that in degoogled OSes the "Use Google Play Location" toggle will be available in the settings regardless of whether there is a play-location service running (doesn't matter if its Google or microG implemented), see https://github.com/organicmaps/organicmaps/pull/9575#issuecomment-2475651608.

I don't think its a blocker, but would be good to detect properly, of course.

A more reliable check would be to try connect to the service and catch exceptions, see https://github.com/organicmaps/organicmaps/pull/9575#issuecomment-2474718719

**Comment by pastk** [comment]
Yes, a proposal is to replace it in the Web version first and give it some time to gather feedback.

**Comment by rtsisyk** [comment]
Make sense. Let's try to enable microG implementation for Web & F-droid, but keep GMS for Google and Huawei.
